### DISCUSSION TITLE
Bilsem ÖYG Ders Notları (Python)

### DISCUSSION THREAD

**Discussion by sonsuzus** [sonsuzus/sonsuzus#446] [Bilsem]
Bu başlık altında her hafta bilsem öyg ders notlarını ve işlediğimiz konuları paylaşacağım. Gelemediğinizde buradan takip edebilir veya takıldığınız yerleri haftanın altına yorum olarak sorabilirsiniz.

**Comment by sonsuzus** [comment]
26 Eylül - 30 Eylül

Watsapp gruba üye olma.
Github hesap açma.
BTK Akademi, Udemy, youtube python dersleri ve kaynakları kullanımı.
Zeka soruları ve algoritma arasındaki bağlantılar.
Python kurulumu (kendi bilgisayarlarınıza)
Online python editörleri ve derleyicileri
Python hakkında genel bilgi.
Python Idle açıp basit işlemler yapabilme.
Ders saatleri düzenleme ve değiştirme.

**Comment by sonsuzus** [comment]
3 Ekim - 7 Ekim

Python progralama dili özellikleri
Shell ve IDLE kullanımları
Yeni dosya oluşturma, kaydetme, çalıştırma
Program nedir? Hangi öğeleri vardır.
Hata ayıklama nedir?
Hata türleri
Hataları çözme
Doğal ve biçimsel diller

**Comment by sonsuzus** [comment]
10 Ekim - 14 Ekim

Python değer ve değişken kavramı
Atama işlemleri
Değişken isim kuralları
Değer tipleri
Anahtar kelimeler
Cümleler ve deyimler
İşlemler
İşlem öncelikleri
Girdi fonksiyonu (input)
Değişken dönüşümleri

**Comment by sonsuzus** [comment]
17 Ekim - 21 Ekim

Kullanıcıya havanın sıcaklığı santigrat cinsinden sorulur ve fahrenayta çevrilir. (F = 9C/5 + 32 )
Kullanıcıya hikayesindeki karakterlerin isimleri sorulur ve o isimlerle hikaye yazdırılır.
Kullanıcıya çap sorulur ve dairenin çevresi ve alanı hesaplanır. (alan = pi r^2) (cevre = 2pi r) (r = R/2)
Kullanıcıya elindeki türk parası miktarı sorulur, gelen yanıta göre ne kadar dolar ve euro alacağı söylenir.
Kullanıcıya saat cinsinden şimdiki zamanı ve kaç saat beklenmesini soran bir program yazınız. Programınız, alarm çaldığında saatin kaç olduğunu gösteren bir çıkış vermelidir.

**Reply by sonsuzus** [reply]
Ödevler:
Kullanıcıya eşkenar üçgenin bir kenarı sorulur gelen yanıta göre eşkenar üçgenin alanı hesaplanır.

Kullanıcıya kürenin çapı sorulur, gelen yanıta göre hacmi hesaplanır.

**Reply by BerraDuruDemir** [reply]
```python
kenar=int(input("kenar uzunluğu nedir"))
uzunluk=int(input("uzunluk nedir"))
alan=((kenar*uzunluk)/2)
print(alan)
```

**Reply by BerraDuruDemir** [reply]
```python
çap=int(input("kürenin çapı nedir"))
r=(çap/2)
pi=(3.14)
hacim=((4*pi*(r**3))/3)
print(hacim)
```

**Reply by MASahin223** [reply]
```python
k = float(input("Eşkenar üçgenin bir kenarını cm cinsinden giriniz: "))
h = 3**0.5*k/2
alan = 1/2*k*h
alan2 = 3**0.5*k**2/4
#İki farklı işlem vardır.
print("Eşkenar üçgenin alanı",alan,"olur.")
print("Eşkenar üçgenin alanı",alan2,"olur.")
```

**Reply by MASahin223** [reply]
```python
#V = 4/3 πr³

R = float(input("Kürenin çapını cm cinsinden giriniz: "))
r = R/2
pi = 3.14
print("Kürenin hacmi",4/3*pi*r**3,"cm küp olur.")
```

**Reply by mervedam** [reply]
```
#alan yaklaşık değeridir kök üç 1.7 alındı
k=int(input("Bir üçgenin kenar uzunluğunu girin\n"))
k2=k*k
kök=1.7
alan=k2*kök/4
print("üçgenin alanı:",alan)
```

**Reply by mervedam** [reply]
```
çap=int(input("kürenin çapı nedir\n"))
pi=3.14
yçap=çap/2
print("kürenin hacmi",4/3*pi*yçap**3,"dır")
```

**Reply by yahyaemiraksoy** [reply]
R = int(input("kürenin çapı kaç cm?"))
r = R/2
pi = 3.14
print(4/3*pi*r**3)

**Reply by AliSelimDemir** [reply]
```
R = int(input("Kürenin çapını giriniz:"))
print("Kürenizin hacmi:", 4/3*3.14*(R/2)**3)
#4/3*pi*r**3
```

**Reply by OmerB2** [reply]
```python
a = float(input("Eşkenar üçgenin kenarını girin:"))
print("Eşkenar üçgenin yaklaşık alanı:",round(a**2*((3/4)**0.5)))
b = float(input("Kürenin çapını girin:"))
print("Kürenin yaklaşık hacmi:",round((4/3)*3.14*(b/2)**3))
```

**Reply by TheEmirAK** [reply]
R = int(input("çap ne"))
r = (R/2)
pi = 3.14
a = ((4*pi)*(r**2))
print ("alan", a)
(öbürünü yapamadım.)

**Reply by yigidi** [reply]
a=float(input("eşkanar üçgenin bir kenarı kaç br: "))

alan=(a**2 * 3**0.5)/4
print("Alan",alan)

**Reply by yigidi** [reply]
import math
R=float(input("Çap kaç? :"))
r = R/2
pi=math.pi
hacim =(4/3) * pi * r**3
print("Kürenin hacmi: ",hacim)

**Reply by K4D1R61** [reply]
Kenar=int(input("Üçgenin kenar uzunluğu nedir"))
Alan=(kenar*2)
print(alan)

**Reply by ArvenAA** [reply]
cap = float(input("Kürenin çapını giriniz "))
    yarıcap = cap/2
    hacim = (4/3)*3.14*yarıcap**3                                                        
    print("Kürenin hacmi",hacim)

**Reply by ArvenAA** [reply]
import math
    kenar = int(input("Üçgenin bir kenarını girin "))
    alan = kenar**2*math.sqrt(3)/4
    print("Üçgeninizin alanı",alan)

**Reply by AliSelimDemir** [reply]
```
celcius = int(input("Hava sıcaklığını giriniz:"))
f = 9celcius/5 + 32
print("Hava sıcaklığı ", f, "fahrenayt derece")
```

**Reply by asyaaqq** [reply]
çap=int(input("kürenin çapı ne"))
yarıçap=(çap/2)
pi=(3.14)
hacmi=((4*pi*(r**3))/3)
print(hacim)

**Comment by omersalih177** [comment]
kenar = int(input(" Kenar ne kadar? "))
uzunluk = int(input( "Uzunluk ne kadar? "))
alan = (taban*yükseklik/2)
print(alan)

**Comment by EMREKAANELMACI** [comment]
kenar=int(input("bir kenar uzunluğu nedir"))
yükseklik=int(input("yükseklik nedir"))
alan=(kenar*yükseklik)/2
print(alan)

**Comment by omersalih177** [comment]
çap = int(input( "Çap nedir? ")
pi = 3.14
r = (çap/2)
hacim = ("(4/3)*(3.14)*(r**3)")
print(hacim)

**Comment by EMREKAANELMACI** [comment]
çap = int(input( "Çap nedir?"))
pi = 3.14
r = (çap/2)
hacim = (4/3)*((pi*(r**3)))
print(hacim)

**Comment by AliSelimDemir** [comment]
````
celcius = int(input("Hava sıcaklığını giriniz:"))
f = 9celcius/5 + 32
print("Hava sıcaklığı ", f, "fahrenayt derece")
```

**Comment by AliSelimDemir** [comment]
```
isim1 = input("Hikayenin ilk karakterinin ismini yazınız lütfen:")
isim2 = input("Hikayenin ikinci karakterinin ismini yazınız lütfen:")
isim3 = input("Hikayenin üçüncü karakterinin ismini yazınız lütfen:")
print("Bir gün okulda", isim1, "ve", isim2, "kavga ediyormuş ancak kavgaya bir çözüm getiremiyorlarmış.", isim1, "yerde gördüğü sayıya 9 diyormuş ancak", isim2, """yer-
deki sayıya 6 diyormuş. Konuya bir çözüm getiremedikleri için""", isim3, "kişisini çağırmışlar yanlarına. ", isim1, "ve", isim2, """kişilerine 'Her zaman kendi açınızdan bakmayın deyip'""", isim1, "ve", isim2, "kişilerinin yerlerini değiştirmiş.")
```

**Comment by AliSelimDemir** [comment]
```
#Kullanıcıya çap sorulur ve dairenin çevresi ve alanı hesaplanır. (alan = pi r^2) (cevre = 2pi r) (r = R/2)
R = int(input("Çapı giriniz lütfen"))
r = R/2
alan = 3.14*r**2
cevre = 2*3.14*r
```

**Comment by AliSelimDemir** [comment]
```
tl = int(input("Kaç paran var fakir?"))
dolar = tl/28
euro = tl/30
print("Bu kadar para ile", dolar, "kadar dolar ve", euro, "kadar euro alabirsin maalesef")

```

**Comment by AliSelimDemir** [comment]
```
simdikizaman = int(input("Şu an saat kaç?"))
alarmsaati = int(input("Alarm kaç saat sonra çalsın?"))
print("Alarm çaldığında saat", (simdikizaman+alarmsaati)%24, "olacak")
```

**Comment by sonsuzus** [comment]
24 Ekim - 28 Ekim

Kullanıcıya çap sorulur, kürenin hacmi hesaplanır.

Kullanıcıya bir kenarın uzunluğu sorulur, eşkenar üçgenin alanı hesaplanır.

print() ve input() fonksiyonları hakkında araştırma yapılacak.
